### PR TITLE
Support tree hierarchy in Available Views view

### PR DESCRIPTION
**Pull request by bhufmann**
It uses the parentId of the OutputDescriptor to group the data providers into a tree.

image

It depends on #1128 and PR to provide the parentId
 
Signed-off-by: Bernd Hufmann <bernd.hufmann@ericsson.com>

### CODE REVIEW

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 1]
```diff
@@ -181,4 +169,67 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
...
+        if (id === undefined) {
+            id = this._idGenerator++;
+            idStringToNodeId[entry.id] = id;
+        }
+        let parentId = -1;
+        if (entry.parentId) {
+            const existing = idStringToNodeId[entry.parentId];
+            if (existing === undefined) {
+                parentId = this._idGenerator++;
+                idStringToNodeId[entry.id] = parentId;
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx:222]: Should this be [entry.parentId] ?

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 2]
```diff
@@ -181,4 +169,67 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
...
+            }
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx:223]: If existing is defined then it should be assigned to parentId?

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 3]
```diff
@@ -181,4 +169,67 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
...
+            lookup[output.id] = node;
+            this._nodeIdToOutput[node.id] = output;
+        });
+        // Create the tree in the order it has been received
+        list.forEach(output => {
+            const node = lookup[output.id];
+            if (output.parentId !== undefined) {
+                const parent: TreeNode = lookup[output.parentId];
+                if (parent) {
+                    if (parent.id !== node.id) {
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx:190]: This protects against the model indicating parentId pointing to the node itself?

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx (hunk 3 above)
Comment (bhufmann) [reply packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx:190]: yes

**Review by PatrickTasse — APPROVED** [review]

**Pull request merged**